source/server/server.cc

```diff
@@ -422,6 +422,9 @@ void InstanceImpl::initialize(const Options& options,
 void InstanceImpl::startWorkers() {
   listener_manager_->startWorkers(*guard_dog_);
   initialization_timer_->complete();
+  // Set the server state as soon as initialization is done.
```

Comment:
Where was this done previously?

Reply:
It is done during flush call - https://github.com/envoyproxy/envoy/blob/master/source/server/server.cc#L203 along with other stats. I think it should continue to be there to report changes if any based on health checks but this call was added so that Envoy reports its readiness at the earliest.

Reply:
What about the other stats at the site you link that are gauges etc? Should they be factored out and we call into that here?

Reply:
I do not think that is required. They can be flushed at the time of flush interval. Since this is used for readiness probe, we want this to be set as soon as Envoy is ready.

Reply:
It would be better to just do this structurally so we don't have to reason on a per-stat basis around what is needed before the first flush interval IMHO

Reply:
Ok. Fair enough, I do not have a strong opinion either way - I refactored it a separate method and called it from both places.